libpromises/locks.c

```diff
@@ -468,14 +468,67 @@ static void PromiseTypeString(char *dst, size_t dst_size, const Promise *pp)
 
 #ifdef __MINGW32__
 
-static bool KillLockHolder(ARG_UNUSED const char *lock)
+// Windows only functions for killing a process
+// NOTE: GetLastError(), FormatMessage(), OpenProcess() and TerminateProcess() are Windows API functions
+
+static void WindowsLogError()
 {
-    Log(LOG_LEVEL_VERBOSE,
-          "Process is not running - ignoring lock (Windows does not support graceful processes termination)");
-    return true;
+    int error_code = GetLastError();
+    Log(LOG_LEVEL_ERR, "Windows API error code: %i", error_code);
+
+    char * error_message = NULL;
+
+    FormatMessage(FORMAT_MESSAGE_FROM_SYSTEM|FORMAT_MESSAGE_ALLOCATE_BUFFER
+                  |FORMAT_MESSAGE_IGNORE_INSERTS,
+                  NULL, error_code,
+                  MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT),
+                  &error_message, 0, NULL);
```

Comment:
I think that `GetLastError()` already calls `FormatMessage` internally. Is there a particular reason you need it here, or can you just use that result?

(I haven't checked the exact calling parameters, I'm not *that* confident in the Win32 API!)